main/src/modules/reactiveLayer/behaviorManager/src/pointing.cpp

```diff
@@ -10,28 +10,59 @@ void Pointing::configure() {
 void Pointing::run(Bottle args/*=Bottle()*/) {
     yInfo() << "Pointing::run";
     Bottle *sensation = sensation_port_in.read();
-    
-    int id = yarp::os::Random::uniform(0, sensation->size()-1);
-    yDebug()<<"Randomly selected: "<< id; // should be random here 
+
+    int id = yarp::os::Random::uniform(0, sensation->size() - 1);
+    yDebug() << "Randomly selected: " << id; // should be random here 
     string obj_name = sensation->get(id).asList()->get(1).asString();
-    
+
+    iCub->opc->checkout();
+    yInfo() << "[pointing] : opc checkout";
+    list<Entity*> lEntities = iCub->opc->EntitiesCache();
+    string aName = "";
+    for (auto& entity : lEntities)
+    {
+        if (entity->entity_type() == "agent")
+        {
+            aName = entity->name();
+        }
+    }
+    if (aName == "")
+    {
+        yInfo() << "No agents detected, looking action aborted.";
+    }
+    else
+    {
+        iCub->look(aName);
+        Time::delay(0.5);
+    }
+
     iCub->say("I could point to the " + obj_name);
     Time::delay(2.0);
-    // yInfo() << "About to point the " + obj_name;
-    // return;
 
-    iCub->say("Do you know that this is a " + obj_name);
     Object* obj = iCub->opc->addOrRetrieveEntity<Object>(obj_name);
-    string sHand = "right";
-    if (obj->m_ego_position[1]<0) 
-        sHand = "left";
+    //string sHand = "right";
+    //if (obj->m_ego_position[1] < 0)
+    //    sHand = "left";
+
+    //Bottle bHand(sHand);
+
+    //bool succeeded = iCub->point(obj_name, bHand);
```

Comment:
Can you please delete these comments? They won't ever be uncommented, and thus just make the code harder to read.